group-coordinator/src/test/java/org/apache/kafka/coordinator/group/consumer/CurrentAssignmentBuilderTest.java

```diff
@@ -0,0 +1,548 @@
+        ));
+
+        ConsumerGroupMember updatedMember = new CurrentAssignmentBuilder(member)
+            .withTargetAssignment(11, targetAssignment)
+            .withCurrentPartitionEpoch((topicId, partitionId) -> {
+                if (topicId.equals(topicId1))
+                    return -1;
+                else
+                    return 10;
+            })
+            .build();
+
+        assertEquals(ConsumerGroupMember.MemberState.ASSIGNING, updatedMember.state());
+        assertEquals(10, updatedMember.previousMemberEpoch());
+        assertEquals(11, updatedMember.memberEpoch());
+        assertEquals(11, updatedMember.nextMemberEpoch());
+        assertEquals(mkAssignment(
+            mkTopicAssignment(topicId1, 3, 4, 5),
+            mkTopicAssignment(topicId2, 6)
+        ), updatedMember.assignedPartitions());
+        assertEquals(Collections.emptyMap(), updatedMember.partitionsPendingRevocation());
+        assertEquals(mkAssignment(
+            mkTopicAssignment(topicId2, 7, 8)
+        ), updatedMember.partitionsPendingAssignment());
+    }
+
+    @Test
+    public void testTransitionFromAssigningToStable() {
+        Uuid topicId1 = Uuid.randomUuid();
+        Uuid topicId2 = Uuid.randomUuid();
+
+        ConsumerGroupMember member = new ConsumerGroupMember.Builder("member")
+            .setMemberEpoch(10)
+            .setPreviousMemberEpoch(11)
+            .setNextMemberEpoch(11)
+            .setAssignedPartitions(mkAssignment(
+                mkTopicAssignment(topicId1, 3),
+                mkTopicAssignment(topicId2, 6)))
+            .setPartitionsPendingAssignment(mkAssignment(
+                mkTopicAssignment(topicId1, 4, 5),
+                mkTopicAssignment(topicId2, 7, 8)))
+            .build();
+
+        assertEquals(ConsumerGroupMember.MemberState.ASSIGNING, member.state());
+
+        Assignment targetAssignment = new Assignment(mkAssignment(
+            mkTopicAssignment(topicId1, 3, 4, 5),
+            mkTopicAssignment(topicId2, 6, 7, 8)
+        ));
+
+        ConsumerGroupMember updatedMember = new CurrentAssignmentBuilder(member)
+            .withTargetAssignment(11, targetAssignment)
+            .withCurrentPartitionEpoch((topicId, partitionId) -> -1)
+            .build();
+
+        assertEquals(ConsumerGroupMember.MemberState.STABLE, updatedMember.state());
+        assertEquals(10, updatedMember.previousMemberEpoch());
+        assertEquals(11, updatedMember.memberEpoch());
+        assertEquals(11, updatedMember.nextMemberEpoch());
+        assertEquals(mkAssignment(
+            mkTopicAssignment(topicId1, 3, 4, 5),
+            mkTopicAssignment(topicId2, 6, 7, 8)
+        ), updatedMember.assignedPartitions());
+        assertEquals(Collections.emptyMap(), updatedMember.partitionsPendingRevocation());
+        assertEquals(Collections.emptyMap(), updatedMember.partitionsPendingAssignment());
+    }
+
+    @Test
+    public void testTransitionFromStableToStable() {
```

Comment:
Is this the right test? Seems like a repeat of the above.

Reply:
Indeed, it does not seem right. Let me fix it.